vendor/sigs.k8s.io/apiserver-network-proxy/konnectivity-client/pkg/client/conn.go

```diff
@@ -113,13 +114,26 @@ func (c *conn) SetWriteDeadline(t time.Time) error {
 // proxy service to notify remote to drop the connection.
 func (c *conn) Close() error {
 	klog.V(4).Infoln("closing connection")
-	req := &client.Packet{
-		Type: client.PacketType_CLOSE_REQ,
-		Payload: &client.Packet_CloseRequest{
-			CloseRequest: &client.CloseRequest{
-				ConnectID: c.connID,
+	var req *client.Packet
+	if c.connID != 0 {
+		req = &client.Packet{
+			Type: client.PacketType_CLOSE_REQ,
+			Payload: &client.Packet_CloseRequest{
+				CloseRequest: &client.CloseRequest{
+					ConnectID: c.connID,
+				},
 			},
-		},
+		}
+	} else {
+		// Never received a DIAL response so no connection ID.
+		req = &client.Packet{
+			Type: client.PacketType_DIAL_CLS,
+			Payload: &client.Packet_CloseDial{
+				CloseDial: &client.CloseDial{
+					Random: c.random,
```

Comment:
is this used anywhere?

Reply:
Yeah, it's used in the proxy server: https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/253/files#diff-104a11712684e520f0affa95e6a053faeb713306a10f69ee99b9729cc9dd1996R448-R454.

Reply:
"Random" is a terrible name. Its essentially an id for the pending dial generated by the KAS. The konnectivity server will pick a final id which is used once the connection is established. In this case we are attempting to shut down a timed out pending connection (which was never established). So the only ID the KAS has to shut down that pending connection is "random".

Reply:
ack, didn't notice this was being sent on the wire to the server... that explains why there was no visible code in this PR making use of it